site/content/integrate/apps/lifecycle/callbacks.md

```diff
@@ -3,5 +3,14 @@ title: Callbacks
 description: Methods that the Apps plugin will invoke on the App
 weight: 10
 ---
-### Coming soon!
-In the future this page will describe the various callback functions that the Apps plugin may call at some point in the App's lifecycle. For example: `OnInstall`, `OnUninstall`.
+The App [manifest]({{<ref "/integrate/apps/structure/manifest">}}) defines a number of callback methods ([Calls]({{<ref "/integrate/apps/structure/call">}})) that are invoked by the Mattermost server at various times. If a callback isn't defined in the manifest, it won't be executed by the Mattermost server.
+The following table lists supported callbacks:
+
+| Name                  | Manifest Parameter       | Details                                                                                                                                                                                                                                                                                                                                                                             |
+|-----------------------|--------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `OnInstall`           | `on_install`             | OnInstall gets invoked when a System Admin installs the App with a `/apps install` command. It may return another call to the app, or a form to display.                                                                                                                                                                                                                            |
+| `OnUninstall`         | `on_uninstall`           | OnUninstall gets invoked when a System Admin uses the `/apps uninstall` command, before the app is actually removed.                                                                                                                                                                                                                                                                |
+| `OnVersionChanged`    | `on_version_changed`     | OnVersionChanged gets invoked when the Mattermost-recommended version of the app no longer matches the previously installed one, and the app needs to be upgraded/downgraded.                                                                                                                                                                                                       |
```

Comment:
2/5 I don't think we should be documenting (and encouraging the use of) this callback. It's ill-defined.

Reply:
Thanks, @levb! Great feedback! Follow-up question: Should this be removed from code as well as the dev docs if ill-defined?

Reply:
I strongly recommend removing this from the Apps framework entirely if it is not implemented or not stable.

Reply:
@neflyte Let's remove it from the docs

Reply:
@levb We probably should remove it from the code if it's not serving a purpose